isisd/isis_pdu.c

```diff
@@ -1783,13 +1780,54 @@ int send_hello_cb(struct thread *thread)
 	int rv = send_hello(circuit, level);
 
 	/* set next timer thread */
-	thread_add_timer(master, send_hello_cb, circuit,
-			 isis_jitter(circuit->hello_interval[level - 1], IIH_JITTER),
-			 &circuit->u.bc.t_send_lan_hello[level - 1]);
-
+	send_hello_sched(circuit, level, 1000 * circuit->hello_interval[level - 1]);
 	return rv;
 }
 
+static void _send_hello_sched(struct isis_circuit *circuit,
+			      struct thread **threadp,
+			      long delay)
+{
+	if (*threadp) {
+		if (thread_timer_remain_msec(*threadp) < (unsigned long)delay)
+			return;
+
+		thread_cancel(*threadp);
+	}
+
+	thread_add_timer_msec(master, send_hello_cb, circuit,
+			      isis_jitter(delay, IIH_JITTER),
+			      threadp);
+}
+
+void send_hello_sched(struct isis_circuit *circuit, int level, long delay)
+{
+	if (circuit->circ_type == CIRCUIT_T_P2P) {
+		_send_hello_sched(circuit, &circuit->u.p2p.t_send_p2p_hello,
+				  delay);
+		return;
+	}
+
+	if (circuit->circ_type != CIRCUIT_T_BROADCAST) {
+		zlog_warn("%s: encountered unknown circuit type %d on %s",
```

Comment:
Question to broader audience, while isis has not been converted to using the new flog_XXX code, should we insist on new warnings/errors be converted now?